src/utils/entitySpec.js

```diff
@@ -180,4 +180,23 @@ export default class EntitySpec {
   toString() {
     return this.toPath()
   }
+
+  static getRevisionToKey(revision, definition) {
+    return definition.provider === 'github' ? revision.sha : revision
+  }
+
+  static getRevisionToString(revision, definition) {
+    return definition.provider === 'github'
+      ? revision.tag === revision.sha
+        ? revision.sha
+        : `${revision.tag} (${revision.sha})`
+      : revision
+  }
+
+  static getEntityName(component) {
+    return component.namespace ? `${component.namespace}/${component.name}` : component.name
+  }
+  static getEntityType(component) {
+    return component.provider === 'github' || component.provider === 'rubygems' ? component.provider : component.type
```

Comment:
don't understand this. type and provider are separate concepts.

Reply:
I saw a lot of methods in the clearlyDefined.js that returns quite the same string.
So what i've done is implement a constant that should help to build the revisions path, and not have a lot of duplicate code: `export const ORIGINS = namespace => `origins/${namespace}` , what i've done is to find a way to build that string in the proper way.
For example, it should be 'origins/github' or 'origins/npm'

This is a use case:
- if i want to get all the versions of `npm/npmjs/-/async/2.6.0` , i should pass to the getRevisions function the type (npm) and the name (async) to return `origins/npm`
- if i want to do the same thing for `git/github/github/fetch/v2.0.3` , i should pass to the same function the provider (github) and the name (async), to return `origins/github`

Maybe the naming is wrong, but i hope the implementation can be clear and helpful

Reply:
I have renamed the function in `getEntityOrigin`, maybe it could be more understandable the purpose?